### PR TITLE
Replace ext/tinycrypt with submodule.

### PR DESCRIPTION
**Pull request by archseer**
`ext/tinycrypt` hasn't been updated in 3 years, and if I remember correctly, I needed https://github.com/intel/tinycrypt/commit/28b33ec081bd24bcaedd93e66ade3432d89ac270 to port mcuboot to a non-gcc compiler/platform.

### CODE REVIEW

**Comment by archseer** [comment]
Seems like the CI runs `git submodule update --init` on the master branch before this PR is included so `ext/tinycrypt` is missing: https://travis-ci.org/github/JuulLabs-OSS/mcuboot/jobs/718507942#L192

**Comment by utzig** [comment]
Bundling `tinycrypt` instead of adding it as a remote was a project decision, so you should instead update it with the latest version (btw, I also don't get why you added two submodules with the same path!).

**Comment by archseer** [comment]
I see, yeah there hasn't been much updates to tinycrypt upstream (there was some minor security related changes though). I might drop this PR completely then -- we're not using tinycrypt currently because we ended up replacing all the crypto functions with hardware accelerated implementations (we followed how the CC310 glue is implemented).

I have a set of patches that extracts the other crypto primitives in a similar way to [sha256](https://github.com/JuulLabs-OSS/mcuboot/blob/5a6e18148d439ae823b53f35c34d51e6df3fdae8/boot/bootutil/include/bootutil/sha256.h) instead of inline ifdefs that I plan on proposing soon as well.

Yeah it was weird to me that the submodule was listed twice, but `sim` didn't compile otherwise. [`mbedtls` is also listed twice on master](https://github.com/JuulLabs-OSS/mcuboot/blob/5a6e18148d439ae823b53f35c34d51e6df3fdae8/.gitmodules#L1-L3)

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [submodule "sim/mbedtls"]
 	path = ext/mbedtls
 	url = https://github.com/ARMmbed/mbedtls
+[submodule "sim/tinycrypt"]
+	path = ext/tinycrypt
```
Comment (de-nordic) [inline .gitmodules:5]: Should it be `sim/tinycrypt`?

**Comment by de-nordic** [comment]
I am not able to clone this branch with submodules. When I try to update all I got is:
```
$$ git submodule update --init --recursive
Submodule path 'boot/cypress/libs/core-lib': checked out '653567762b7aba1ef6e11a1b604cb3bd6b0a9d9b'
Submodule path 'boot/cypress/libs/pdl/psoc6pdl': checked out '0e38b78ef9ae3ce5db0563bdcde80a028c5fc38a'
Submodule path 'boot/cypress/libs/psoc6hal': checked out 'a0d6c08f1178750e29967999860485878fac09f9'
Submodule path 'boot/cypress/libs/retarget-io': checked out 'bd88e66fc6976de49694382a5ce5dc7613f12ace'
Submodule path 'ext/cddl_gen': checked out '9d911cf0c7c9f13b5a9fdd5ed6c1012df21e5576'
Submodule path 'ext/mbedtls': checked out '74692aeb8c7e2c1c0f273592d87472a6f2aabacf'
```

**Comment by de-nordic** [comment]
Sync does not help:
```
$$ git submodule sync --recursive
Synchronizing submodule url for 'boot/cypress/libs/core-lib'
Synchronizing submodule url for 'boot/cypress/libs/cy-mbedtls-acceleration'
Synchronizing submodule url for 'boot/cypress/libs/pdl/psoc6pdl'
Synchronizing submodule url for 'boot/cypress/libs/psoc6hal'
Synchronizing submodule url for 'boot/cypress/libs/retarget-io'
Synchronizing submodule url for 'ext/cddl_gen'
Synchronizing submodule url for 'ext/mbedtls'
Synchronizing submodule url for 'ext/mbedtls/crypto'
```

File: .gitmodules (hunk 1 above)
Comment (archseer) [reply .gitmodules:5]: https://github.com/JuulLabs-OSS/mcuboot/pull/785#issuecomment-675348710

**Comment by archseer** [comment]
Closing since not using submodules was a design decision

**Pull request closed without merging**